### ISSUE TITLE
panic when trying to compile non existent file

### ISSUE THREAD

**Issue by yoav-steinberg** [winglang/wing#1007]
### I tried this

```sh
cd libs/wingc
cargo build && ../../target/debug/wingc i_dont_exist.w
```

### Expected

An error message about file not being found and a 1 exit code

### Instead, this happened

```
thread 'main' panicked at 'Error reading source file: ../../tmp.w: Os { code: 2, kind: NotFound, message: "No such file or directory" }', libs/wingc/src/lib.rs:44:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

### Component

Compiler

### Anything else?

This should be easy to fix, just return on that error from _lib.rs_'s `parse()` function and fill the `Diagnostics` accordingly.

**Comment by msalman-abid** [comment]
@yoav-steinberg this can be marked as done now

**Comment by staycoolcall911** [comment]
Closed by #1063

**Comment by staycoolcall911** [comment]
@msalman-abid , thanks for tying all these loose ends.

By the way, to close an issue automatically using the relevant PR, all you need to do is include the following text in the PR description: "fixes #xxx" (where xxx is the issue number).
